src/Functions/array/arrayElement.cpp

```diff
@@ -819,11 +820,49 @@ void FunctionArrayElement::executeMatchKeyToIndex(
     const Offsets & offsets, PaddedPODArray<UInt64> & matched_idxs, const Matcher & matcher)
 {
     size_t rows = offsets.size();
-    for (size_t i = 0; i < rows; ++i)
+    size_t expected_match_pos = 0;
+    bool matched = false;
+    if (rows)
```

Comment:
```suggestion
    if (rows == 0)
        return;
    size_t expected_match_pos = 0;
    bool matched = false;
```
I think early return is more preferable here. Please also update the loop below, unfortunately I couldn't add a suggestion on github as a single block.